### DISCUSSION TITLE
Can't loft to/from a Face?

### DISCUSSION THREAD

**Discussion by tve** [sgenoud/replicad#87] [Q&A] [answered]
I'm using a finder to identify the face of a Shape and I'd to loft from there to a sketch. But I'm failing...
See [example](https://studio.replicad.xyz/workbench?code=UEsDBAoAAAAIAOw49lZNOm7GTAEAAEwCAAAHAAAAY29kZS5qc22SUUvDMBSF3%252Fsrrn0xha7dFF8EESbsVR8EN8WHLL1dw7KbmqZuo%252By%252Fe9NJ6cSHQNKe75xz0ypLjQcHD%252BCwNlrJIorwUFvnocBStsZD2ZLy2hLspCZRJ9BFAKrn1vYQyGwntziXDc7tQdxOU5jxupkmmXeSGiM9rsTkLokGzjsd5MwS7plfSIULTQU6kWSaXowkFPHyPU6BXUp%252BIzgqha4l%252FdXiPRu0eErGfq%252B27qsUTu6ftFMGxYxZlgAMNUTfLckgz8HII1jyFpYrqENgL2226FX1fFFhwqOE7nk%252BpHHUOTMztvRv2lfid6Ze6dC3juCjt%252BygqWTNpfsRSO54GzcG8RvZXFljHT%252FYOESK4ZReMueUAePjiFmbFv9FQpExE84jzmExYHneQaU3leHlw3cYGfzRh1srLDZ0zT%252BHBUlHX2naXD2y0Wd0in4AUEsBAhQACgAAAAgA7Dj2Vk06bsZMAQAATAIAAAcAAAAAAAAAAAAAAAAAAAAAAGNvZGUuanNQSwUGAAAAAAEAAQA1AAAAcQEAAAAA)
where I'm trying to loft from the red face to the blue circle.

![image](https://github.com/sgenoud/replicad/assets/39480/1962c35b-254d-4504-a665-89ec99fc9486)

Uncomment line 11 (the loftWith) to see the error I'm getting:
```
const r = replicad

export default function main(p) {
  const box = r.makeBaseBox(30, 10, 20).translateY(-5)

  const triBase = new r.FaceFinder().inPlane("XZ", 0).find(box, {unique: true})
  const triTop = r.drawCircle(10).
    translate(0, 10). // lay onto XY plane
    sketchOnPlane("XZ", -20)

  //const tri = triTop.loftWith(triBase)

  return [
    { shape: box, name: "sleeve", color: "green" },
    { shape: triTop, name: "tri", color: "blue" },
    { shape: triBase, name: "triBase", color: "red" },
    //{ highlightFace: triBase, color: "red" }, // doesn't do anything!?
  ]
}

```

**Comment by sgenoud** [comment, marked as answer]
You need to loft with another sketch, and not a face (which might have hole in it and become more complicated.

You can see how I would do this [here](https://studio.replicad.xyz/workbench?code=UEsDBAoAAAAIAGFf%252FlaVKrjxnwEAABYDAAAHAAAAY29kZS5qc22SXUvDMBSG7%252FsrDrtKYXZO8Wag4AcDryYo%252BIVgTE%252FXYJbU00QnY%252F%252Fdk2x2nQgtJOl53rznPVXOth4IToGwMVrJMstw2TjyUGIlg%252FFQBau8dhYWUlvR5LDKAFTi3twyksVCvuOFbPHCLcXx4RDG%252FB4d5oUnaVsjPT6Kg5M86zhPOpYza%252FGL%252BalUONW2RBJ5oe2NkRbF4OFpMARWqfiL4KuGsIJg9UfACSsEhHXeV7xzTTJTkvy61KQMijHTXALQGRHJXV7AaARGfoOz3sHDIzTxylTavqNX9WzPxAE3E90zdA1t7YIpIabELfhatyD5gRpNgwSvG4HY0muXXeez4uPbVNA1v9mKbSaFMo7vzQsXPNK9Jl7HxpO37Uiu%252BDTJTn6TLKyjhTTnfieDlnlG%252B%252BCM9Fz3qE1RLFnvT4fNbRLt7BhX%252BXvta7HrICGEPpCF53TNisORDY8nTcvKBS8HrUH8RE5ROeOID%252BaEaHlf63lt%252BPUxqwmIKofTM6j%252B%252FAA8LVgP9%252BXZWyfP6572mwk4%252BK8%252B9ttn4r7HEZZb7CVbZz9QSwECFAAKAAAACABhX%252F5WlSq48Z8BAAAWAwAABwAAAAAAAAAAAAAAAAAAAAAAY29kZS5qc1BLBQYAAAAAAQABADUAAADEAQAAAAA%253D)

I should add the `sketchFace` helper method.

Also, you can see how you would highlight a face (but I need to document this properly).
